third_party/amd/lib/TritonAMDGPUToLLVM/BuiltinFuncToLLVM.cpp

```diff
@@ -98,6 +111,44 @@ class CallOpConversion : public mlir::RewritePattern {
     rewriter.replaceOp(callOp, loadVal);
     return mlir::success();
   }
+
+  LogicalResult convertToLLVMIntrinsic(LLVM::CallOp callOp,
+                                       mlir::PatternRewriter &rewriter) const {
+    StringRef calleeName = callOp.getCallee().value();
+
+    auto operands = callOp.getOperands();
+    auto result = callOp.getResult();
+
+    LLVM::LLVMFunctionType calleeType = callOp.getCalleeType().value();
+    Type returnType = calleeType.getReturnType();
+
+    LLVM::FastmathFlagsAttr flags;
+    auto loc = callOp.getLoc();
+
+    Operation *replacementOp = nullptr;
+    if (calleeName == "__ocml_trition_abs") {
+      assert(operands.size() == 1);
+      if (type::isInt(returnType)) {
+        replacementOp = rewriter.create<LLVM::AbsOp>(
+            loc, returnType, operands[0], /*is_int_min_poison=*/false);
+      } else if (type::isFloat(returnType)) {
+        replacementOp =
+            rewriter.create<LLVM::FAbsOp>(loc, returnType, operands[0], flags);
+      }
+    } else if (calleeName == "__ocml_trition_llrint") {
+      assert(operands.size() == 1);
+      // Note, LrintOp and LlrintOp result in a code-gen error
+      Operation *op = rewriter.create<LLVM::RintOp>(loc, operands[0].getType(),
+                                                    operands[0]);
+      replacementOp =
+          rewriter.create<LLVM::FPToSIOp>(loc, returnType, op->getResult(0));
+    }
+
+    if (replacementOp)
+      rewriter.replaceOp(callOp, replacementOp);
+
```

Comment:
Return failure if no replacementOp?

Reply:
I have no idea why github doesn't show the code update. The current solution is as follows

```cpp
    if (replacementOp) {
      rewriter.replaceOp(callOp, replacementOp);
      return mlir::success();
    }

    return mlir::failure();
```